### PR TITLE
Binary Search Tree requires enumerator

### PR DESCRIPTION
**Pull request by remcopeereboom**
In the Ruby core classes `#each` returns an `Enumerator` to itself if it has not gottten a block to work on.

I added a test that requires this and updated the example to work with the new tests.

Added a version constant for the tests.

### CODE REVIEW

File: binary-search-tree/example.rb [hunk 1]
```diff
@@ -1,7 +1,10 @@
 class Bst
+  VERSION = 2
```
Comment (kotp) [inline binary-search-tree/example.rb]: Version 1 would suffice, prior version was 0, as it was not tracked for versions.

**Comment by remcopeereboom** [comment]
Decremented version number to use zero-based indexing :)

**Comment by kotp** [comment]
This pull review is really out of date, it is 'pre-exercise folder' out of date.

**Comment by remcopeereboom** [comment]
That seems to be an issue with exercism. I only forked minutes earlier. I remember this having occured before. Is there a way to rebase right on github?

**Comment by kotp** [comment]
I just did a `git rebase origin/master` while on the pull request branch... it did have white space errors but otherwise was simple to rebase.

The white space errors will still need to be addressed, other than that, this should end up being a 'clean' operation.

**Comment by remcopeereboom** [comment]
> I just did a git rebase origin/master while on the pull request branch... it did have white space errors but otherwise was simple to rebase.

Thanks, but that's not what I meant. I meant is there a way to do the rebase while on github.com, i.e. without needing to download the repository to my computer. I often only fork, change from within github, and then pull request. Typically the time is so short that there is no need to rebase. It would be so convenient if I could also do the rebasing online so I can work on computers that don't have `git` installed.

**Comment by remcopeereboom** [comment]
I'll resubmit.

**Comment by kytrinyx** [comment]
I don't think there is a way to rebase from GitHub.

**Comment by remcopeereboom** [comment]
No I didn't think so, but thanks for mentioning it!

**Pull request closed without merging**